x/gamm/pool-models/stableswap/amm.go

```diff
@@ -38,10 +38,6 @@ func cfmmConstant(xReserve, yReserve osmomath.BigDec) osmomath.BigDec {
 // We use this version for calculations since the u
 // term in the full CFMM is constant.
 func cfmmConstantMultiNoV(xReserve, yReserve, wSumSquares osmomath.BigDec) osmomath.BigDec {
-	if !xReserve.IsPositive() || !yReserve.IsPositive() || wSumSquares.IsNegative() {
-		panic("invalid input: reserves must be positive")
-	}
```

Comment:
I think this check is worth keeping – this module deals with enough math where ensuring basic invariants are never broken (even if some checks are superfluous) is well worth it

Reply:
I understand, but cfmmConstantMultiNoV first calls cfmmConstantMultiNoVY which has the same checks at the beginning.

Reply:
Fair enough! I'm happy with removing then :)